bc_obps/common/middleware/mock_time_middleware.py

```diff
@@ -0,0 +1,31 @@
+# type: ignore
+from datetime import datetime
+from typing import Callable
+
+from django.conf import settings
+from django.http import HttpRequest, HttpResponse
+import time_machine
+
+
+class MockTimeMiddleware:
+    def __init__(self, get_response: Callable[[HttpRequest], HttpResponse]):
+        self.get_response = get_response
+
+    def __call__(self, request: HttpRequest) -> HttpResponse:
+
+        # Mocking time before processing
+        if settings.ENVIRONMENT in ["local", "dev", "test"] and 'mock-time' in request.COOKIES:
+            time_traveller = time_machine.travel(datetime.fromisoformat(request.COOKIES['mock-time']))
+            request.time_traveller = time_traveller
```

Comment:
Necessary to attach the time_traveller instance to the request, otherwise we run into threading issues